ext/strscan/strscan.c

```diff
@@ -1458,6 +1458,56 @@ strscan_fixed_anchor_p(VALUE self)
     return p->fixed_anchor_p ? Qtrue : Qfalse;
 }
 
+typedef struct {
+    VALUE self;
+    VALUE captures;
+} named_captures_data;
+
+static int
+named_captures_iter(const OnigUChar *name,
+                    const OnigUChar *name_end,
+                    int back_num,
+                    int *back_refs,
+                    OnigRegex regex,
+                    void *arg)
+{
+    named_captures_data *data = arg;
+
+    VALUE key = rb_str_new((const char *)name, name_end - name);
+    VALUE value = RUBY_Qnil;
+    int i;
+    for (i = 0; i < back_num; i++) {
+        value = strscan_aref(data->self, INT2NUM(back_refs[i]));
+    }
+    rb_hash_aset(data->captures, key, value);
+    return 0;
+}
+
+/*
+ * call-seq:
+ *   scanner.named_captures -> hash
+ *
+ * Returns a hash of string variables matching the regular expression.
+ *
+ *   s = 'foobarbaz'
+ *   re = /(?<f>foo)(?<r>bar)(?<z>baz)/
+ *   scan = StringScanner.new(s)
+ *   scan.match? re
```

Comment:
We can simplify this with literal.

```suggestion
 *   scan = StringScanner.new('foobarbaz')
 *   scan.match?(/(?<f>foo)(?<r>bar)(?<z>baz)/)
```